kore/src/Kore/Step/Simplification/And.hs

```diff
@@ -179,7 +179,11 @@ makeEvaluateNonBool notSimplifier sideCondition patterns = do
             let (term1, condition1) = Pattern.splitTerm pattern1
             unified <- termAnd notSimplifier term1 term2
             pure (Pattern.andCondition unified condition1)
-    unified <- Foldable.foldlM unify Pattern.top (term <$> patterns)
+    unified <-
+        Foldable.foldlM
+            unify
+            Pattern.top
+            (term <$> Foldable.toList patterns)
```

Comment:
```suggestion
            (MultiAnd.map term patterns)
```

Reply:
If I remember correctly this will make some tests fail. I believe it's because there are patterns with the same term and MultiAnd will remove the duplicates. For some reason this seemed reasonable to me, but now that I think about it more maybe I should investigate and see if there's something else going wrong.

Reply:
I took another look at this and the unit test would fail because the resulting pattern would have sort `PREDICATE_` instead of the right sort. This is because when `MultiAnd` is `\top`, we lose all sort information.